routes/api/report.js

```diff
@@ -1,10 +1,41 @@
 'use strict';
 
-exports.track = {
+const getDaysAndHours = (paramString) => {
+  const day = new RegExp('(\\d+)(d)').exec(paramString);
+  const hour = new RegExp('(\\d+)(h)').exec(paramString);
+  const retObj = {};
+  if (hour) {
+    retObj.hour = hour[1];
+  }
+  if (day) {
+    retObj.day = day[1];
+  }
+  return retObj;
```

Comment:
make this function return back the start date since you are already doing the if statement for hours days